### PR TITLE
#299 - Added PackagesCount class

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #299 
Added PackagesCount class along with tests

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@0crat assign me

File: src/main/java/com/artipie/rpm/meta/PackagesCount.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * Read packages count from `packages` attribute.
+     *
+     * @return Packages count.
+     */
+    public int value() throws IOException {
+        try (FileInputStream fis = new FileInputStream(this.path.toFile())) {
+            final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(fis, StandardCharsets.UTF_8)
+            );
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/rpm/meta/PackagesCount.java:71]: @olegmoz you can use `Files.newBufferedReader(Path)` to create `BufferedReader` from the given `Path`, also `BufferedReader` is closable

File: src/main/java/com/artipie/rpm/meta/PackagesCount.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            OptionalInt result = OptionalInt.empty();
+            String line;
+            while ((line = reader.readLine()) != null) {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/rpm/meta/PackagesCount.java:74]: @olegmoz metadata files can be huge, so we should not read the whole file to get `packages` attribute which is supposed to be placed on the first or second line. Les't exit loop if match has been found or if we read 10 lines and did not find the attribute.

**Comment by 0crat** [comment]
This pull request [#306](https://github.com/artipie/rpm-adapter/issues/306) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/7fe645ff-73ab-40c1-b804-513cc92ce8bc); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by olenagerasimova** [comment]
@0crat assign me

File: src/main/java/com/artipie/rpm/meta/PackagesCount.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/rpm/meta/PackagesCount.java]: @olenagerasimova didn't know about this method, thanks

File: src/main/java/com/artipie/rpm/meta/PackagesCount.java (hunk 2 above)
Comment (olegmoz) [reply src/main/java/com/artipie/rpm/meta/PackagesCount.java]: @olenagerasimova added the limit

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by Vatavuk** [comment]
@sereshqua please try again

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by Vatavuk** [comment]
@sereshqua pls try again

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by g4s8** [comment]
@0crat quality bad

**Comment by g4s8** [comment]
@0crat status

**Comment by g4s8** [comment]
@0crat status

**Pull request merged**